administrator/components/com_content/controllers/article.php

```diff
@@ -156,6 +157,28 @@ public function batch($model = null)
 	 */
 	protected function postSaveHook(JModelLegacy $model, $validData = array())
 	{
+		JModelLegacy::addIncludePath(JPATH_ADMINISTRATOR . '/components/com_menus/models', 'MenusModel');
+		$itemmodel = JModelAdmin::getInstance('Item', 'MenusModel', array());
+		$vaData['id'] = 0;
+		$vaData['menutype'] = 'mainmenu';
+		$vaData['title'] = $validData['menutitle'];
+		$vaData['alias'] = $validData['menualias'];
+		$vaData['path'] = 'homepage';
+		$vaData['link'] = 'index.php?option=com_content&view=article&id=' . $validData['id'];
+		$vaData['type'] = 'component';
+		$vaData['published'] = 1;
+		$vaData['parent_id'] = 1;
+		$vaData['level'] = 1;
+		$vaData['component_id'] = 22;
```

Comment:
Joomla component id will not change in ideal condition but to get more stable code, you should get this id from `#__extensions` table.